docs/testnet/tutorial.md

```diff
@@ -0,0 +1,158 @@
+
+Our variables should now look like this
+![Step 1 screenshot](https://images.tango.us/workflows/dc61d575-7eea-457c-bdfe-8edccf79b366/steps/3d9b387b-a492-4096-aa8a-1d50084be0d5/6d08adce-70d2-4203-8d18-c09999536553.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=413%3A131)
+
+
+Next, let's think about how we'll instantiate this contract. We'll need to:
+- create an instance of the ERC-20 contract to manage and hold OGG tokens
+- generate a secret random number
+
+
+Let's extend to the constructor to include these:
+
+![Step 2 screenshot](https://images.tango.us/workflows/a8278591-dc09-4e26-97a2-b30776f86179/steps/58146527-d6d4-47c4-a796-cc0b3b4909db/56afa096-9f71-49a0-a85d-94ba3ecc56a4.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=410%3A117)
+
+
+Now let's write our function to generate the random number and secretly store it within the contract
+
+![Step 1 screenshot](https://images.tango.us/workflows/a8278591-dc09-4e26-97a2-b30776f86179/steps/8b18e51b-b4da-43e6-a752-eea0cc550791/009e7455-a298-4b66-808b-39080f30af24.png?crop=focalpoint&fit=crop&fp-x=0.5000&fp-y=0.5000&fp-z=1.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=864%3A211)
+
+
+This isn't great as although the number is arguably random, it's not random to anyone who knows how it's generated and how to measure the inputs when the function is called. We are working on something better. Watch this space!
+
+Okay, so right now, we should have the following:
+- a constructor that initializes the contract
+- a way to calculate the random number to be guessed
+
+We now need to add the ability for a player to make an actual guess. But before we do that, we have not thought about the range we might want numbers to appear in. This would allow us to set the difficulty for any particular game. So let's do that now.
+
+Extend the constructor to accept a range int and create a variable for it, so we have something that looks like this
+
+![Step 1 screenshot](https://images.tango.us/workflows/8f7ef3ef-091f-4b5d-9017-d041449139ae/steps/43aff627-6025-42c1-8e1a-654761d01b90/c3fc2de4-e9fa-4d31-b1a8-abe5331e0ec9.png?crop=focalpoint&fit=crop&fp-x=0.2957&fp-y=0.3023&fp-z=4.0000&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=2307%3A1244)
+
+And we can change our random number function to adhere to the range by including the modulo operator %, like this:
```

Comment:
modulo?

Reply:
It gives you the remainder when one number is divided by another